src/core/qgsattributetableconfig.cpp

```diff
@@ -242,6 +248,12 @@ Qt::SortOrder QgsAttributeTableConfig::sortOrder() const
 
 void QgsAttributeTableConfig::setSortOrder( Qt::SortOrder sortOrder )
 {
+  // fix https://hub.qgis.org/issues/15803
+  if ( sortOrder != Qt::AscendingOrder && sortOrder != Qt::DescendingOrder )
```

Comment:
If an invalid value is specified, does it make more sense to force the value to ascending or to keep the current one?

i.e. what do you think about

```
if ( sortOrder != Qt::AscendingOrder && sortOrder != Qt::DescendingOrder )
    return;
```